### PR TITLE
`next start` on a build built with a different Next.js version will error [13954]

### LINKED ISSUE #13954: Running `next start` on a build built with a different Next.js version should error
# Bug report

## Describe the bug

Currently, you can run `next build` and `next start` on different versions of Next.js. This causes unexpected or broken behavior.

## Expected behavior

`next start` should refuse to boot the application if the version does not match what `next build` was.

### PR DESCRIPTION
**Pull request by darshkpatel**
Fixes: #13954

Added nextVersion: string to ```pages-manifest.json``` generated by webpack to track which version of next.js was used for the build.

This is used to compare versions and throw an error if versions differ

@timneutkens added the discussed changes in 0a81e69

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| buildDuration | 13.1s | 13.2s | ⚠️ +64ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ⚠️ +2.1 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.242 | 2.259 | ⚠️ +0.02 |
| / avg req/sec | 1115.19 | 1106.57 | -8.62 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.545 | 1.494 | -0.05 |
| /error-in-render avg req/sec | 1618.25 | 1672.8 | ⚠️ +54.55 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| _error.js gzip | 3.37 kB | 3.37 kB | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| Overall change | 8.34 kB | 8.34 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| _error.module.js gzip | 2.21 kB | 2.21 kB | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.49 kB | 5.49 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 544 B | 544 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| index.html gzip | 956 B | 956 B | ✓ |
| link.html gzip | 963 B | 963 B | ✓ |
| withRouter.html gzip | 949 B | 949 B | ✓ |
| Overall change | 2.87 kB | 2.87 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| buildDuration | 13.9s | 13.8s | -52ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ⚠️ +2.1 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| _error.js gzip | 3.37 kB | 3.37 kB | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| Overall change | 8.34 kB | 8.34 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| _error.module.js gzip | 2.21 kB | 2.21 kB | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.49 kB | 5.49 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 544 B | 544 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ⚠️ +12 B |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ⚠️ +12 B |
| link.js | 913 kB | 913 kB | ⚠️ +12 B |
| routerDirect.js | 905 kB | 905 kB | ⚠️ +12 B |
| withRouter.js | 905 kB | 905 kB | ⚠️ +12 B |
| Overall change | 4.48 MB | 4.48 MB | ⚠️ +60 B |

</details>

</details>

**Comment by ijjk** [comment]
## Failing test suites 

**test/integration/404-page/test/index.test.js** 
- 404 Page Support > server mode > should add /404 to pages-manifest correctly
- 404 Page Support > serverless mode > should add /404 to pages-manifest correctly

<details>
<summary>Expand output</summary>

  ● 404 Page Support › server mode › should add /404 to pages-manifest correctly

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      70 |         join(appDir, '.next', mode, 'pages-manifest.json')
      71 |       )
    > 72 |       expect('/404' in manifest).toBe(true)
         |                                  ^
      73 |     })
      74 |   }
      75 | }

      at Object.<anonymous> (integration/404-page/test/index.test.js:72:34)

  ● 404 Page Support › serverless mode › should add /404 to pages-manifest correctly

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      70 |         join(appDir, '.next', mode, 'pages-manifest.json')
      71 |       )
    > 72 |       expect('/404' in manifest).toBe(true)
         |                                  ^
      73 |     })
      74 |   }
      75 | }

      at Object.<anonymous> (integration/404-page/test/index.test.js:72:34)

</details>

**Comment by darshkpatel** [comment]
On it, updating the remaining tests

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| buildDuration | 12.2s | 12s | -293ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ⚠️ +2.1 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.007 | 2.025 | ⚠️ +0.02 |
| / avg req/sec | 1245.58 | 1234.79 | -10.79 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.209 | 1.152 | -0.06 |
| /error-in-render avg req/sec | 2067.15 | 2169.66 | ⚠️ +102.51 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| _error.js gzip | 3.37 kB | 3.37 kB | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| Overall change | 8.34 kB | 8.34 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| _error.module.js gzip | 2.21 kB | 2.21 kB | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.49 kB | 5.49 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 544 B | 544 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| index.html gzip | 956 B | 956 B | ✓ |
| link.html gzip | 963 B | 963 B | ✓ |
| withRouter.html gzip | 949 B | 949 B | ✓ |
| Overall change | 2.87 kB | 2.87 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| buildDuration | 13.2s | 13.1s | -113ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ⚠️ +2.1 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.51 kB | 6.51 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.6 kB | 5.6 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| _error.js gzip | 3.37 kB | 3.37 kB | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| Overall change | 8.34 kB | 8.34 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| _error.module.js gzip | 2.21 kB | 2.21 kB | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.49 kB | 5.49 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 544 B | 544 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ⚠️ +12 B |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ⚠️ +12 B |
| link.js | 913 kB | 913 kB | ⚠️ +12 B |
| routerDirect.js | 905 kB | 905 kB | ⚠️ +12 B |
| withRouter.js | 905 kB | 905 kB | ⚠️ +12 B |
| Overall change | 4.48 MB | 4.48 MB | ⚠️ +60 B |

</details>

</details>

File: packages/next/build/webpack/plugins/pages-manifest-plugin.ts [hunk 1]
```diff
@@ -18,8 +22,11 @@ export default class PagesManifestPlugin implements Plugin {
   apply(compiler: Compiler): void {
     compiler.hooks.emit.tap('NextJsPagesManifest', (compilation) => {
       const entrypoints = compilation.entrypoints
-      const pages: PagesManifest = {}
-
+      const nextVersion: string = process.env.__NEXT_VERSION ?? ''
```
Comment (timneutkens) [inline packages/next/build/webpack/plugins/pages-manifest-plugin.ts:25]:
```suggestion
      const nextVersion: string = process.env.__NEXT_VERSION
```

File: packages/next/next-server/server/next-server.ts [hunk 2]
```diff
@@ -196,7 +196,12 @@ export default class Server {
     const pagesManifestPath = join(this.serverBuildDir, PAGES_MANIFEST)
 
     if (!dev) {
-      this.pagesManifest = require(pagesManifestPath)
+      this.pagesManifest = require(pagesManifestPath) as PagesManifest
+      if (this.pagesManifest.nextVersion !== process.env.__NEXT_VERSION) {
+        throw new Error(
```
Comment (timneutkens) [inline packages/next/next-server/server/next-server.ts:201]: I think it'd be better to do a warning instead to make sure this doesn't break existing users

Comment (darshkpatel) [reply packages/next/next-server/server/next-server.ts]: Done! 8c764ec

File: packages/next/build/webpack/plugins/pages-manifest-plugin.ts (hunk 1 above)
Comment (darshkpatel) [reply packages/next/build/webpack/plugins/pages-manifest-plugin.ts]: Done! 8c764ec

**Comment by darshkpatel** [comment]
Whoops! Looking into it

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| buildDuration | 10s | 10.8s | ⚠️ +761ms |
| nodeModulesSize | 66.1 MB | 66.1 MB | ⚠️ +3.08 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.597 | 1.87 | ⚠️ +0.27 |
| / avg req/sec | 1565.2 | 1336.57 | ⚠️ -228.63 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 0.945 | 1.09 | ⚠️ +0.15 |
| /error-in-render avg req/sec | 2646.87 | 2292.78 | ⚠️ -354.09 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..18e4.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.71 kB | 5.71 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.09 kB | 7.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.4 kB | 26.4 kB | ✓ |
| Overall change | 26.4 kB | 26.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| index.html gzip | 954 B | 954 B | ✓ |
| link.html gzip | 958 B | 958 B | ✓ |
| withRouter.html gzip | 945 B | 945 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| buildDuration | 11.1s | 10.7s | -415ms |
| nodeModulesSize | 66.1 MB | 66.1 MB | ⚠️ +3.08 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..18e4.js gzip | 10.7 kB | 10.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.71 kB | 5.71 kB | ✓ |
| webpack-HASH..dule.js gzip | 751 B | 751 B | ✓ |
| 19b7e98f51cc..dule.js gzip | 7.09 kB | 7.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.7 kB | 52.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.4 kB | 26.4 kB | ✓ |
| Overall change | 26.4 kB | 26.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 268 B | 268 B | ✓ |
| _buildManife..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 540 B | 540 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | MLH-Fellowship/next.js darsh/14017 | Change |
| - | - | - | - |
| _error.js | 876 kB | 876 kB | ⚠️ +12 B |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 876 kB | 876 kB | ⚠️ +12 B |
| link.js | 915 kB | 915 kB | ⚠️ +12 B |
| routerDirect.js | 908 kB | 908 kB | ⚠️ +12 B |
| withRouter.js | 908 kB | 908 kB | ⚠️ +12 B |
| Overall change | 4.49 MB | 4.49 MB | ⚠️ +60 B |

</details>

</details>
Commit: 1a4fb2f8e7a0e4d0df7d46abad6660fc8a9f812b

**Comment by timneutkens** [comment]
Going to close this as it's no longer super relevant and has merge conflicts. Thanks for the PR!

**Pull request closed without merging**